pkg/k8s/watchers/secret.go

```diff
@@ -165,3 +167,16 @@ func k8sToEnvoySecret(secret *slim_corev1.Secret) *envoy_entensions_tls_v3.Secre
 func getEnvoySecretName(namespace, name string) string {
 	return fmt.Sprintf("%s/%s", namespace, name)
 }
+
+func uniq(arr []string) []string {
+	keys := make(map[string]bool)
+	for _, entry := range arr {
+		keys[entry] = true
+	}
+
+	var list []string
```

Comment:
Nit: `list := make([]string, 0, len(keys))`

Reply:
somehow i always forget about this :(, thanks thanks :100: